config/fields/date.php

```diff
@@ -46,42 +46,24 @@
         /**
          * Latest date, which can be selected/saved (Y-m-d)
          */
-        'max' => function (string $max = null) {
-            return $this->toDatetime($max);
+        'max' => function (string $max = null): ?string {
+            return Date::optional($max);
         },
         /**
          * Earliest date, which can be selected/saved (Y-m-d)
          */
-        'min' => function (string $min = null) {
-            return $this->toDatetime($min);
+        'min' => function (string $min = null): ?string {
+            return Date::optional($min);
         },
 
         /**
          * Round to the nearest: sub-options for `unit` (day) and `size` (1)
          */
         'step' => function ($step = null) {
-            $default = [
-                'size' => 1,
-                'unit' => 'day'
-            ];
-
-            if ($step === null) {
-                return $default;
-            }
-
-            if (is_array($step) === true) {
-                $step = array_merge($default, $step);
-                $step['unit'] = strtolower($step['unit']);
-                return $step;
-            }
-
-            if (is_int($step) === true) {
-                return array_merge($default, ['size' => $step]);
-            }
-
-            if (is_string($step) === true) {
-                return array_merge($default, ['unit' => strtolower($step)]);
-            }
+            return Date::stepConfig($step, [
+                'size' => 5,
+                'unit' => 'minute',
+            ]);
```

Comment:
Default step for the date field should be 1 day. Only when `time` option is active, default step should be 5 minutes.